auth/src/main/java/com/firebase/ui/auth/ui/account_link/WelcomeBackPasswordPrompt.java

```diff
@@ -114,6 +113,7 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
 
     private void next(String email, final String password) {
         final FirebaseAuth firebaseAuth = mActivityHelper.getFirebaseAuth();
+        setIntent(getIntent().putExtras(mActivityHelper.getMergeFailedIntent()));
```

Comment:
It can't be used anywhere after `signInWithEmailAndPassword` because then I wouldn't be saving the old uid.